src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -1005,7 +1005,7 @@ public static Task WriteAllLinesAsync(string path, IEnumerable<string> contents,
                 int index = 0;
                 while (index < count)
```

Comment:
Could be done separately.  I simply want to fix the bug here with the most concise fix possible.